ecommerce/baskets/views.py

```diff
@@ -15,12 +20,45 @@ class BasketItemViewSet(DetailedViewSetMixin, viewsets.ModelViewSet):
         "detailed": BasketItemDetailedSerializer,
     }
 
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        user_id = self.request.user.id
+        # import pdb; pdb.set_trace()
+        return queryset.filter(basket__customer__id=user_id)
+
 
 class BasketViewSet(DetailedViewSetMixin, viewsets.ModelViewSet):
-    queryset = Basket.objects.all()
-    serializer_class = BasketSerializer
-    filterset_class = BasketFilter
+    http_method_names = ["get", "delete"]
+    permission_classes = ()
+    queryset = BasketItem.objects.all()
+    serializer_class = BasketItemDetailedSerializer
+    filterset_class = BasketItemFilter
     serializer_action_classes = {
-        "detailed_list": BasketDetailedSerializer,
-        "detailed": BasketDetailedSerializer,
+        "add_to_cart": BasketItemSerializer,
     }
+
+    def get_queryset(self):
+        queryset = super().get_queryset()
+        user_id = self.request.user.id
+        return queryset.filter(basket__customer__id=user_id)
+
+    @action(detail=False, methods=["post"], http_method_names=["post"])
+    def add_to_cart(self, request, pk=None):
+        user_id = self.request.user.id
+        product = Product.objects.get(id=request.data["product"])
+        quantity = request.data["quantity"]
+        price = product.price
+        basket = Basket.objects.filter(customer__id=user_id, status="open").first()
+        if not basket:
+            basket = Basket.objects.create(customer_id=user_id, status="open")
+        basket_item = BasketItem.objects.filter(
+            basket__customer__id=user_id, product=product, price=float(str(price))).first()
```

Comment:
Endpoint'den price değerini 55.15 olarak girersem sanırım float(str(price)) değeri 55.1 olarak geliyor bende. 2 decimal olduğunda price=float(str(price)) eşitliği sağlanamıyor.

Reply:
price değerini elle girmiyorum. add to cart kısmında product ve quantity alanı var. price değerine otomatik olarak productın price değeri olarak atanıyor.